src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHashShared.cs

```diff
@@ -0,0 +1,784 @@
+// Copyright (c) Alexandre Mutel. All rights reserved.
+// Licensed under the BSD-Clause 2 license.
+// See license.txt file in the project root for full license information.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+
+namespace System.IO.Hashing
+{
+    /// <summary>Shared implementation of the XXH3 hash algorithm for 64-bit in <see cref="XxHash3"/> and <see cref="XxHash128"/> version.</summary>
+#if NET5_0_OR_GREATER
+    [SkipLocalsInit]
```

Comment:
Isn't this attribute applied to the entire module?

Reply:
Apparently not.

![image](https://user-images.githubusercontent.com/12659251/200129596-811890e8-7c5b-4fcc-ba28-4c5d3ce42ccb.png)

Reply:
It's a pseudo attribute baked into individual methods' IL streams. It's enabled by default across projects as needed and there's no need to specify it explicitly.

Reply:
I see `.locals init` in `System.IO.Hashing v7.0.0-rc.2` when targeting .NET 7.

![image](https://user-images.githubusercontent.com/12659251/200132915-6611b0de-e427-4ccf-8a94-2fc1a2d13d98.png)

Reply:
> It's enabled by default across projects as needed and there's no need to specify it explicitly.

Only in the shared framework right now. We could remove that condition if desirable.